libraries/bot-azure/src/test/java/com/microsoft/bot/azure/CosmosDbPartitionStorageTests.java

```diff
@@ -181,7 +220,88 @@ public void HandleCrazyKeys() {
         super.handleCrazyKeys(storage);
     }
 
+    // NOTE: THESE TESTS REQUIRE THAT THE COSMOS DB EMULATOR IS INSTALLED AND STARTED !!!!!!!!!!!!!!!!!
+    @Test
+    public void WaterfallCosmos() {
+        ConversationState convoState = new ConversationState(storage);
+
+        ConversationReference conversationReference = TestAdapter.createConversationReference("waterfallTest", "User1", "Bot");
+        TestAdapter adapter = new TestAdapter(conversationReference).use(new AutoSaveStateMiddleware(convoState));
+
+        StatePropertyAccessor<DialogState> dialogState = convoState.createProperty("dialogStateForWaterfallTest");
+        DialogSet dialogs = new DialogSet(dialogState);
+
+        dialogs.add(new TextPrompt("TextPrompt", new PromptValidator<String>() {
+            public CompletableFuture<Boolean> promptValidator(PromptValidatorContext<String> promptContext) {
+                String value = promptContext.getRecognized().getValue();
+                if (value.length() > 3) {
+                    StringBuilder sb = new StringBuilder("You got it at the ").append(promptContext.getAttemptCount()).append("th try!");
+                    Activity succeededMessage = MessageFactory.text(sb.toString());
+                    return promptContext.getContext().sendActivities(succeededMessage).thenApply(resourceResponses -> true);
+                }
+
+                Activity reply = MessageFactory.text("Please send a name that is longer than 3 characters. " + promptContext.getAttemptCount());
+                return promptContext.getContext().sendActivities(reply).thenApply(resourceResponses -> false);
```

Comment:
In C# this is [sendActivity](https://github.com/microsoft/botbuilder-dotnet/blob/main/tests/Microsoft.Bot.Builder.Azure.Tests/CosmosDbPartitionStorageTests.cs#L220). Is this okay?